### PR TITLE
Datadog metrics

### PR DESCRIPTION
**Pull request by joel-hamill**
## Description
- https://jira.mesosphere.com/browse/DCOS_OSS-727
- https://github.com/mesosphere/dcos-docs-enterprise/pull/1228

## Urgency
- [ ] Blocker 
- [ ] High
- [x] Medium

## Requirements
- Test all commands and procedures.
- Build content [locally](https://github.com/dcos/dcos-docs#test-local) and test for formatting/links.
- Add redirects to [dcos-website/redirect-files](https://github.com/dcos/dcos-website#managing-redirects).
- Change all affected versions (e.g. 1.7, 1.8, and 1.9).
- See the [contribution guidelines](https://github.com/dcos/dcos-docs#contributing).

### CODE REVIEW

**Comment by joel-hamill** [comment]
@darkonie i wasn't able to test this since i do not have a Datadog API key.

File: 1.9/metrics/datadog.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+## Install the Datadog agent in your DC/OS cluster
+
+Install the `datadog` package in DC/OS:
+
+1.  Go to the **Universe** tab of the DC/OS GUI and find the **Datadog** package. 
+    ![datadog package](/docs/1.9/img/datadog-package.png)
+1.  Click **INSTALL PACKAGE** -> **ADVANCED INSTALLATION** and enter [your Datadog API_KEY](https://app.datadoghq.com/account/settings#api).
+1.  Click **REVIEW AND INSTALL** to complete your installation.
+
+After a minute or two a Datadog agent will be running in the cluster at `datadog-agent.marathon.mesos:8125`. This is the default location used by the Datadog plugin.
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: comma after "two"

File: 1.9/metrics/datadog.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+## Test the DC/OS Datadog metrics plugin (agents only)
+As a stopgap during testing, you may be able to manually run the Datadog plugin on your agents by running it as a Marathon task. You must first upload your binary to a web server that's visible to your cluster, then create a Marathon application like the following (with customized `cmd`, `instances`, and `uris` to meet your needs):
+
+```json
+{
+  "cmd": "chmod +x ./dcos-metrics-* && ./dcos-metrics-* -dcos-role agent -auth-token <CONTENT OF 'dcos config show core.dcos_acs_token'>",
+  "instances": NUMBER_OF_AGENTS,
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: number of agents should be `<number-of-agents>`

File: 1.9/metrics/datadog.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  "id": "test-datadog-plugin",
+  "cpus": 0.1,
+  "mem": 128,
+  "disk": 0,
+  "acceptedResourceRoles": [ "slave_public", "*" ]
+}
+```
+
+### Install the DC/OS Datadog metrics plugin
+When you're happy with the test results, you'll need to install the plugin into your cluster. For each host in your cluster, transfer your binary for the plugin and then add a systemd unit to manage the service. This unit differs slightly between agent and master hosts.
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: How does the testing work? What am I looking for?

File: 1.9/metrics/datadog.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+### Deploy the Metrics Plugin to Every Cluster Host
+
+1.  Copy the Datadog plugin from a local host to your remote host (`my.host:/usr/bin`):
+
+    ```bash
+    scp dcos-metrics-datadog-plugin-1.0.0-rc7 my.host:/usr/bin
+    ```
+    
+    
+1.  [SSH to your master node] and assign permission to the plugin.
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: "assign permissions" (it's more than one permission you're assigning)

File: 1.9/metrics/datadog.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+```
+[Unit]
+Description=DC/OS Datadog Metrics Plugin (master)
+
+[Service]
+ExecStart=/usr/bin/dcos-metrics-datadog-plugin-1.0.0rc7 -dcos-role master -metrics-port 80 -auth-token <MY_AUTH_TOKEN>
+```
+
+### Agent Systemd Unit
+Add a agent systemd unit file on your master node and save as `/etc/systemd/system/dcos-metrics-datadog-plugin.service`.
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: "Add an"

File: 1.9/metrics/datadog.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+### Enable, Start, and Verify
+
+1.  Enable and start the Datadog plugin.
+
+    ```bash
+    systemctl enable dcos-metrics-datadog-plugin && systemctl start dcos-metrics-datadog-plugin
+    ```
+    
+1.  View the system logs and verify that everything is okay.
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: how do I know everything is OK?

Comment (joel-hamill) [inline 1.9/metrics/datadog.md]: @philipnrmn ^

File: 1.9/metrics/datadog.md (hunk 3 above)
Comment (joel-hamill) [inline 1.9/metrics/datadog.md]: @philipnrmn ^

File: 1.9/metrics/datadog.md (hunk 6 above)
Comment (philipnrmn) [inline 1.9/metrics/datadog.md]: There should be no failure to start - ie the service is actively producing logs which are not failures from systemd. Additionally, there should be no error-level logs. Here's an example of error-level log:
```
May 05 09:03:57 ip-10-0-2-9.us-west-2.compute.internal dcos-metrics[32138]: time="2017-05-05T09:03:56Z" level=error msg="json: unsupported value: NaN" collector=framework
```

File: 1.9/metrics/datadog.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+The Datadog metrics plugin for DC/OS supports sending metrics from the DC/OS metrics service on master and agent hosts to a Datadog agent for shipping to DatadogHQ.
+
+# Installation
+
+## Build the plugin
+
+**Prerequisite:** 
+- DC/OS is installed
+- [Go programming environment](https://golang.org/doc/install) <!-- dcos-metrics must be run from within the go directory -->
+- GitHub
```
Comment (sascala) [inline 1.9/metrics/datadog.md]: what aspect of GitHub is a prerequisite? Does the user need a GitHub account? I guess they also need git installed on their machine? Would a link be helpful?

Comment (joel-hamill) [inline 1.9/metrics/datadog.md]: @sascala https://github.com/dcos/dcos-docs/pull/968/commits/6ae9560c3363f0a3588f5e0de1c1373e29e21bbc

**Comment by joel-hamill** [comment]
@sascala is this ready to merge?

**Comment by sascala** [comment]
@joel-hamill There was one outstanding question and some clarifications about testing, but I don't think they're _crucial_

File: 1.9/metrics/datadog.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+Install the `datadog` package in DC/OS:
+
+1.  Go to the **Universe** tab of the DC/OS GUI and find the **Datadog** package. 
+    ![datadog package](/docs/1.9/img/datadog-package.png)
+1.  Click **INSTALL PACKAGE** -> **ADVANCED INSTALLATION** and enter [your Datadog API_KEY](https://app.datadoghq.com/account/settings#api).
+1.  Click **REVIEW AND INSTALL** to complete your installation.
+
+After a few minutes, a Datadog agent will be running in the cluster at `datadog-agent.marathon.mesos:8125`. This is the default location used by the Datadog plugin.
+
+## Test the DC/OS Datadog metrics plugin (agents only)
```
Comment (joel-hamill) [inline 1.9/metrics/datadog.md]: @philipnrmn is the rest of the information (lines 57-137) all testing, or is testing just lines 57-70? Please clarify.

CC: @sascala

Comment (philipnrmn) [inline 1.9/metrics/datadog.md]: Testing is just lines 57-70.

File: 1.9/metrics/datadog.md (hunk 3 above)
Comment (philipnrmn) [inline 1.9/metrics/datadog.md]: The only reason to do this step (unless you're a plugin developer) is to check that metrics are being piped to DataDog via the agent as you expect. So 'success' would just be data showing up in your dashboard.

**Pull request merged**